File: python/src/Behavioral/Observer/DvdSubscriber.py
#//DvdSubscriber- the observer
class DvdSubscriber:

		def __init__ (self,subscriberName) :
			self.subscriberName = subscriberName

		def __str__(self) :
			return '"subscriberName":"{0}"'.format(self.getSubscriberName())

		def getSubscriberName (self) :
			return self.subscriberName

		def newDvdRelease (self,newDvdRelease,subscriptionListName) :
			return """Hello {0} , subscriber to the  {1} DVD release list.
The new Dvd {2} will be released on {3}/{4}/{5}.
""".format(self.getSubscriberName(), subscriptionListName, newDvdRelease.dvdName, newDvdRelease.dvdReleaseMonth, newDvdRelease.dvdReleaseDay, newDvdRelease.dvdReleaseYear)

		def updateDvdRelease (self,newDvdRelease, subscriptionListName) :
			return """Hello {0} subscriber to the {1} DVD release list.
The following DVDs release has been revised: {2} will be released on {3}/{4}/{5}
""".format(self.getSubscriberName(), subscriptionListName, newDvdRelease.dvdName,  newDvdRelease.dvdReleaseMonth, newDvdRelease.dvdReleaseDay, newDvdRelease.dvdReleaseYear)

File: ruby/src/Behavioral/Observer/DvdSubscriber.rb
#//DvdSubscriber- the observer
class DvdSubscriber
		attr_accessor :subscriberName

		def initialize(subscriberName)
			@subscriberName = subscriberName
		end

		def to_s()
 			return "\"subscriberName\":\"#{getSubscriberName()}\""
		end

		def getSubscriberName ()
 			return @subscriberName
		end

		def newDvdRelease (newDvdRelease,subscriptionListName)
 			return "Hello #{ getSubscriberName() }, subscriber to the #{ subscriptionListName } DVD release list.\nThe new Dvd #{ newDvdRelease.dvdName } will be released on #{ newDvdRelease.dvdReleaseMonth }/#{ newDvdRelease.dvdReleaseDay }/#{ newDvdRelease.dvdReleaseYear }.\n"
		end

		def updateDvdRelease (newDvdRelease, subscriptionListName)
 			return "Hello #{ getSubscriberName() }, subscriber to the #{ subscriptionListName } DVD release list.\nThe following DVDs release has been revised: #{ newDvdRelease.dvdName  } will be released on #{ newDvdRelease.dvdReleaseMonth }/#{ newDvdRelease.dvdReleaseDay }/#{ newDvdRelease.dvdReleaseYear }.\n"
		end
end

